modules/oe_content_social_media_field/tests/Kernel/SocialMediaFieldTest.php

```diff
@@ -0,0 +1,228 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\Tests\oe_content\Kernel;
+
+use Drupal\Core\Entity\Entity\EntityViewDisplay;
+use Drupal\field\Entity\FieldConfig;
+use Drupal\field\Entity\FieldStorageConfig;
+use Drupal\KernelTests\Core\Entity\EntityKernelTestBase;
+use Drupal\node\Entity\Node;
+use Drupal\node\Entity\NodeType;
+
+/**
+ * Tests the social media field type definition.
+ */
+class SocialMediaFieldTest extends EntityKernelTestBase {
+
+  /**
+   * A field storage to use in this test class.
+   *
+   * @var \Drupal\field\Entity\FieldStorageConfig
+   */
+  protected $fieldStorage;
+
+  /**
+   * The field used in this test class.
+   *
+   * @var \Drupal\field\Entity\FieldConfig
+   */
+  protected $field;
+
+  /**
+   * The display options to use in the formatter.
+   *
+   * @var array
+   */
+  protected $displayOptions = [
+    'type' => 'social_media_link_formatter',
+    'label' => 'hidden',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  public static $modules = [
+    'field',
+    'node',
+    'oe_content_social_media_field',
+    'system',
+    'text',
+    'user',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp(): void {
+    parent::setUp();
+
+    $this->installSchema('user', 'users_data');
+    $this->installEntitySchema('user');
+    $this->installEntitySchema('node');
+    $this->installConfig(['field', 'node', 'system']);
+
+    // Create content type.
+    $type = NodeType::create(['name' => 'Test content type', 'type' => 'test_ct']);
+    $type->save();
+
+    $this->fieldStorage = FieldStorageConfig::create([
```

Comment:
Don't we already create a field storage that we ship with? Can't we use that?

Reply:
Since that was not made for testing purposes I prefer the strictly define the test field.